lib/tools.js

```diff
@@ -2329,6 +2329,65 @@ function pipeLinewise(input, output) {
     rl.on('line', line => output.write(line + os.EOL));
 }
 
+/**
+ * Find the adaper main file as full path
+ *
+ * @memberof tools
+ * @param {string} adapter - adapter name of the adapter, e.g. hm-rpc
+ * @returns {Promise<string>}
+ */
+async function getAdapterMainFile(adapter) {
+    const adapterDir = getAdapterDir(adapter);
+    if (!adapterDir) {
+        return Promise.reject(new Error(`Could not find adapter dir of ${adapter}`));
```

Comment:
async function, just throw the error.

Reply:
All cases shouldn't make a difference, but okay, we can go with throw and return w/o explicitly using Promise syntax